### PR TITLE
NCI-Agency/anet#375: Upgrade to React 16

### PR DESCRIPTION
**Pull request by maradragan**

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -80,19 +82,22 @@
...
+    "react-dom": "^16.2.0",
+    "react-redux": "^5.0.7",
+    "react-router": "^4.2.0",
+    "react-router-bootstrap": "^0.24.4",
+    "react-router-dom": "^4.0.0",
     "react-scrollspy": "^3.3.4",
-    "react-tag-input": "^4.8.2",
+    "react-tag-input": "^5.0.1",
+    "redux": "^3.7.2",
+    "sourcemap-lookup": "^0.0.3",
```
[the remark this answers is not in the document]
Comment (gjvoosten) [reply client/package.json]: Probably added by accident (manually, when trying to find a source line reported in the developer console).

[the remark this answers is not in the document]
Comment (maradragan) [reply client/package.json]: Removed it.

File: client/package.json [hunk 2]
```diff
@@ -61,17 +61,19 @@
     "webpack-merge": "^4.1.1"
   },
   "dependencies": {
-    "alloyeditor": "^1.4.1",
+    "alloyeditor": "^1.5.1",
     "bootstrap": "^3.3.7",
     "change-case": "^3.0.1",
     "core-js": "^2.5.3",
+    "create-react-class": "^15.6.3",
```
Comment (VassilIordanov) [inline client/package.json]: where is this used?

File: client/config/webpack.common.js [hunk 3]
```diff
@@ -57,12 +58,26 @@ module.exports = {
...
                     }
                 ]
+            }, {
+                test: /alloy-editor-no-react\.js$/,
+                use: {
+                    loader: 'imports-loader',
+                    options: {
+                        'React': 'react',
+                        'React.PropTypes': 'prop-types',
+                        'React.createClass': 'create-react-class'
```
Comment (VassilIordanov) [inline client/config/webpack.common.js]: I assume a dependency needs this? Could you please document which one in a comment?

Comment (gjvoosten) [reply client/config/webpack.common.js]: `git blame client/config/webpack.common.js`:
```
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 61)             }, {
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 62)                 test: /alloy-editor-no-react\.js$/,
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 63)                 use: {
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 64)                     loader: 'imports-loader',
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 65)                     options: {
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 66)                         'React': 'react',
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 67)                         'React.PropTypes': 'prop-types',
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 68)                         'React.createClass': 'create-react-class'
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 69)                     }
696a4cb1 (Gertjan van Oosten 2018-04-03 16:31:32 +0200 70)                 }
```
`git log 696a4cb1`:
```
commit 696a4cb1fb0324982ed84b7f940b64ecb2ffe18b
Author: Gertjan van Oosten <gertjan@west.nl>
Date:   Tue Apr 3 16:31:32 2018 +0200

    NCI-Agency/anet#375: Re-add Alloy Editor
    
    And rewrite it such that the proper defines are there;
    rewrite is loosely based on https://github.com/liferay/alloy-editor/issues/770#issuecomment-355055340
```

**Pull request merged**